### PR TITLE
chore: add pytorch-lightning to docs requirements

### PR DESCRIPTION
**Pull request by neilconway**
PTL is now required to build the docs.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

**Comment by mackrorysd** [comment]
This may be dependent on https://github.com/determined-ai/determined/pull/2109 to fully fix things: apparently this can trigger a circular dependency because of Tensorboard. That said, I'm not sure what workflow actually breaks and `make all` is currently broken, so I'll +1 this to unblock and ping Hamid.

**Review by mackrorysd — APPROVED** [review]

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,9 +1,12 @@
 # pin docutils at 0.15.2 to avoid transitive dependency conflict with botocore (requires < 0.16)
 docutils==0.15.2
-sphinx==2.4.4
-sphinx-argparse>=0.2.5
-git+https://github.com/determined-ai/determined_sphinx_theme.git@9edfa7c6ce6926def9fc69b8ddd7666f3419a907
-sphinx-sitemap>=2.2.0
+
 pillow
+pytorch-lightning==1.2.4
```
Comment (hamidzr) [inline docs/requirements.txt:5]: our lightning support is based off of `1.2.0` can we have the same version here? should help with the API when developing harness.

**Comment by ioga** [comment]
works for me on macos 👍

**Review by hamidzr — APPROVED** [review]
LGTM

> This may be dependent on #2109 to fully fix things: apparently this can trigger a circular dependency because of Tensorboard. That said, I'm not sure what workflow actually breaks and make all is currently broken, so I'll +1 this to unblock and ping Hamid.

I was suspecting this run into the issue @mackrorysd  was describing in a local build but this + #2074 's move to tf2 look like to be all we need.

Thanks for fixing this.

File: docs/requirements.txt (hunk 1 above)
Comment (neilconway) [reply docs/requirements.txt:5]: 1.2.1 through 1.2.4 are just "standard weekly patch releases" and mostly include bug fixes (https://github.com/PyTorchLightning/pytorch-lightning/releases). Happy to use `1.2.0` for now but IMO hardcoding a single specific minor version of PTL to support is probably not going to be tenable.

**Pull request merged**